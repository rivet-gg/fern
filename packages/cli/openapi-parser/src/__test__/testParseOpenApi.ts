import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
import { CONSOLE_LOGGER } from "@fern-api/logger";
import { serialization } from "@fern-api/openapi-ir-sdk";
import { createMockTaskContext } from "@fern-api/task-context";
import { parse, SpecImportSettings } from "../parse";

const FIXTURES_PATH = join(AbsoluteFilePath.of(__dirname), RelativeFilePath.of("fixtures"));

// eslint-disable-next-line jest/no-export
export function testParseOpenAPI(
    fixtureName: string,
    openApiFilename: string,
    asyncApiFilename?: string,
    settings?: SpecImportSettings
): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("parse open api", async () => {
            const absolutePathToOpenAPI = join(
                FIXTURES_PATH,
                RelativeFilePath.of(fixtureName),
                RelativeFilePath.of(openApiFilename)
            );
            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;
            const specs = [];
            if (absolutePathToOpenAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToOpenAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const openApiIrJson = await serialization.OpenApiIntermediateRepresentation.jsonOrThrow(openApiIr, {
                skipValidation: true
            });
            expect(openApiIrJson).toMatchSnapshot();
        });
    });
}
