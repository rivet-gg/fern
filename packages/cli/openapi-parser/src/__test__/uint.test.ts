import { testParseOpenAPI } from "./testParseOpenApi";

describe("open api parser", () => {
    testParseOpenAPI("uint", "openapi.yml");
});
