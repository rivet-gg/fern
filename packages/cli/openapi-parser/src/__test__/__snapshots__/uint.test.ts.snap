// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`open api parser uint parse open api 1`] = `
{
  "apiVersion": null,
  "basePath": null,
  "channel": [],
  "description": null,
  "endpoints": [
    {
      "audiences": [],
      "authed": false,
      "availability": null,
      "description": null,
      "errors": {},
      "examples": [
        {
          "codeSamples": [],
          "description": null,
          "headers": [],
          "name": null,
          "pathParameters": [],
          "queryParameters": [],
          "request": {
            "properties": {},
            "type": "object",
          },
          "response": {
            "properties": {
              "id": {
                "type": "primitive",
                "value": {
                  "type": "float",
                  "value": 1.1,
                },
              },
              "value": {
                "type": "primitive",
                "value": {
                  "type": "uint64",
                  "value": 1000000,
                },
              },
            },
            "type": "object",
          },
          "type": "full",
        },
      ],
      "generatedRequestName": "GetRequest",
      "headers": [],
      "idempotent": null,
      "internal": null,
      "method": "GET",
      "operationId": "Get",
      "pagination": null,
      "path": "/get",
      "pathParameters": [],
      "queryParameters": [],
      "request": {
        "additionalProperties": false,
        "contentType": null,
        "description": null,
        "schema": {
          "availability": null,
          "description": null,
          "generatedName": "GetRequest",
          "groupName": null,
          "nameOverride": null,
          "schema": "Request",
          "type": "reference",
        },
        "type": "json",
      },
      "requestNameOverride": null,
      "response": {
        "description": "A simple API response.",
        "responseProperty": null,
        "schema": {
          "availability": null,
          "description": null,
          "generatedName": "GetResponse",
          "groupName": null,
          "nameOverride": null,
          "schema": "Response",
          "type": "reference",
        },
        "type": "json",
      },
      "sdkName": null,
      "server": [],
      "summary": null,
      "tags": [],
    },
  ],
  "globalHeaders": [],
  "groups": {},
  "hasEndpointsMarkedInternal": false,
  "idempotencyHeaders": [],
  "nonRequestReferencedSchemas": [
    "Response",
  ],
  "schemas": {
    "Request": {
      "additionalProperties": false,
      "allOf": [],
      "allOfPropertyConflicts": [],
      "availability": null,
      "description": "A generic request type used throughout the API.
",
      "generatedName": "Request",
      "groupName": null,
      "nameOverride": null,
      "properties": [
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "requestId",
          "key": "id",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "requestId",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "RequestId",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "type": "float",
              },
              "type": "primitive",
            },
          },
        },
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "requestValue",
          "key": "value",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "requestValue",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "RequestValue",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "type": "uint64",
              },
              "type": "primitive",
            },
          },
        },
      ],
      "type": "object",
    },
    "Response": {
      "additionalProperties": false,
      "allOf": [],
      "allOfPropertyConflicts": [],
      "availability": null,
      "description": "A generic response type used throughout the API.
",
      "generatedName": "Response",
      "groupName": null,
      "nameOverride": null,
      "properties": [
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "responseId",
          "key": "id",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "responseId",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "ResponseId",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "type": "float",
              },
              "type": "primitive",
            },
          },
        },
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "responseValue",
          "key": "value",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "responseValue",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "ResponseValue",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "type": "uint64",
              },
              "type": "primitive",
            },
          },
        },
      ],
      "type": "object",
    },
  },
  "securitySchemes": {},
  "servers": [],
  "tags": {
    "orderedTagIds": null,
    "tagsById": {},
  },
  "title": "Test spec for \`uint\` types specified as number formats.",
  "variables": {},
  "webhooks": [],
}
`;
